File: core/src/main/java/net/year4000/mapnodes/game/regions/AbstractRegion.java
/*
 * Copyright 2016 Year4000. All Rights Reserved.
 */

package net.year4000.mapnodes.game.regions;

import net.year4000.utilities.Utils;

/** The abstract region that handles the creation of points */
public abstract class AbstractRegion implements Region {
    @Override
    public boolean equals(Object other) {
        return Utils.equals(this, other);
    }

    @Override
    public int hashCode() {
        return Utils.hashCode(this);
    }

    @Override
    public String toString() {
        return Utils.toString(this);
    }
}

File: core/src/main/js/net/year4000/mapnodes/js/regions/abstract_region.js
/*
 * Copyright 2017 Year4000. All Rights Reserved.
 */

/** The abstract region that handles the creation of points */
class AbstractRegion {

  /** Get a immutable set of points the region contains, or if its not possible return empty */
  get points() {
    return Immutable.Set.of()
  }

  /** Check when the vector is in the region */
  contains(vector3) {
    return this.points.has(vector3)
  }

  /** Are both regions equal */
  equals(other) {
    return _.isEqual(this, other)
  }
}
